packages/SwingSet/test/vat-admin/test-create-vat.js

```diff
@@ -290,25 +290,27 @@ test('createVat holds refcount', async t => {
   // if/when things like that change, this test also asserts ko27, but that
   // can be updated in a single place.
 
-  // 'held' is exported by v1, which shows up in the c-lists but doesn't
+  // 'held' is exported by v6, which shows up in the c-lists but exports don't
   // count towards the refcount
   let expectedRefcount = 0;
-  let expectedCLists = 1; // v1-export-held
+  let expectedCLists = 1; // v6-export-held
 
   // bootstrap() imports 'held', adding it to the v2-bootstrap c-list and
   // incrementing the refcount.
-  expectedRefcount += 1; // v2-bootstrap
-  expectedCLists += 1; // v2-bootstrap
+  expectedRefcount += 1; // v1-bootstrap holds resolution of 'createHeld'
+  expectedCLists += 1; // v1-bootstrap
 
   // calling getHeld doesn't immediately increment the refcount
   const kpid1 = c.queueToVatRoot('bootstrap', 'getHeld', []);
   await c.run();
+  expectedRefcount += 1; // from 'getHeld'
```

Comment:
Why did this change? I need to investigate more. XXX-@warner

Reply:
I tracked this one down: the test was buggy. At this point in the execution, there should be *three* references to the "getHeld" (`ko27`) object: one from v1-bootstrap, one from `kp45` (the result promise known as `kpid1`, which we got back from the `c.queueToVatRoot(.., 'getHeld'`), and the "external" one which `kpResolution()` adds to each slot in the resolution value. The first two were applied within delivery cranks, so their updates were committed through the crankbuffer already. But the `kpResolution()` "external" ref is still sitting in the crankbuffer. I instrumented `kernelKeeper.setRefCount()` and it shows the `ko27` count being set to `3,3`, but the test's `findRefs()` reports back `2,2`: it's fetching the stale value from `kvStore`, which will be updated the next time a `commitCrank()` happens .

By adding a `kernelKeeper.commitCrank()` to the end of `kpResolution()`, I was able to make the original test fail, and adding your extra `expectedRefCount +=1` made it pass again.

In your new version, there is no crankbuffer, so when `kpResolution()` increments the refcount, the test can see it immediately.

I'll write up a patch for the test (as a PR aimed at your branch) to update the comments in that test and explain what's happening.

Reply:
https://github.com/Agoric/agoric-sdk/pull/6705